File: leetcode2/0108. Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        if not nums:
            return None
        k=len(nums)//2
        root=TreeNode(nums[k])
        root.left=self.sortedArrayToBST(nums[:k])
        root.right=self.sortedArrayToBST(nums[k+1:])
        return root

File: archives/C#/0108. Convert Sorted Array to Binary Search Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode SortedArrayToBST(int[] nums) {
        int right=nums.Length-1;
        int left=0;
        return subSortedArrayToBST(nums,left,right);
    }
    public TreeNode subSortedArrayToBST(int[] nums,int left,int right){
        if (left>right){
            return null;
        }
        int mid=(left+right)/2;
        TreeNode node=new TreeNode(nums[mid]);
        node.left=subSortedArrayToBST(nums,left,mid-1);
        node.right=subSortedArrayToBST(nums,mid+1,right);
        return node;
        
    }
}
